src/include/sof/math/exp_fcn_hifi4.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_EXPONENTIAL_HIFI4_H__
+#define __SOF_MATH_EXPONENTIAL_HIFI4_H__
+
+#include <stdint.h>
+#include <stddef.h>
+
+#if defined(__XCC__)
+/* HiFi */
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI4 == 1
+#define EXPONENTIAL_HIFI4 1
+
+#else
+#error "No XCHAL_HAVE_HIFIx found. Cannot build XCC EXP - HiFi module."
```

Comment:
#error causes build with TGL HiFi3 to fail. I think the EXPONENTIAL_GENERIC should be set here to build generic version with HiFi3 compiler.